### ISSUE TITLE
Qgis 3.10.0 Conversion from QgsPoinxXY to QgsPoint Fails (Python API)

### ISSUE THREAD

**Issue by matheusfillipe** [qgis/QGIS#32503]
Hi! On my plugin: https://github.com/matheusfillipe/Topografia I often used  the form of conversion: p2=QgsPoint(p) where p is a QgsPointXY type. Recently updated from 8.3 (Where this worked fine) to 10.0 and now that conversion throws an expection:

```
SystemError: returned a result with an error set 
TypeError: must be real number, not QgsPointXYThe above exception was the direct cause of the following exception:
Traceback (most recent call last):  
File "C:/Users/LucasVinícius/AppData/Roaming/QGIS/QGIS3\profiles\default/python/plugins\Topografia-master\app\controller\estacas.py", line 603, in new    id_estaca, table = self.model.new(dist, estaca, lyr, filename) if not isGeopckg else self.model.newEmpty(Config.instance().DIST,filename)  
File "C:/Users/LucasVinícius/AppData/Roaming/QGIS/QGIS3\profiles\default/python/plugins\Topografia-master\app\model\estacas.py", line 51, in new    self.table = self.create_estacas()  
File "C:/Users/LucasVinícius/AppData/Roaming/QGIS/QGIS3\profiles\default/python/plugins\Topografia-master\app\model\estacas.py", line 654, in create_estacas    for vertice, ponto_anterior, qtd_estacas, progressiva, az, sobra, tamanho_da_linha, cosa, cosb, tipo, feature, resto in self.gera_vertice(isCalc):
  File "C:/Users/LucasVinícius/AppData/Roaming/QGIS/QGIS3\profiles\default/python/plugins\Topografia-master\app\model\estacas.py", line 585, in gera_vertice    ponto_inicial = QgsPoint(seg_start)
SystemError:  returned a result with an error set
Versão do Python: 3.7.0 (v3.7.0:1bf9cc5093, Jun 27 2018, 04:59:51) [MSC v.1914 64 bit (AMD64)] Versão do QGIS: 3.10.0-A CoruÃ±a A CoruÃ±a, 6c816b4204
```

By the way I noticed that doing   p2=QgsPoint(p.x(), p.y()) works fine. Is that a bug or this capability was removed? Wouldn't make much sense if that was intentionally removed that's why I'm reporting this.

That was tested both on the linux and the windows version.

**Comment by matheusfillipe** [comment]
This can also be reproduced from the python shell: 

`
pp 
<QgsPointXY: POINT(1 1)>
p=QgsPoint(pp)
TypeError: must be real number, not QgsPointXY

The above exception was the direct cause of the following exception:

Traceback (most recent call last):
  File "/usr/lib/python3.6/code.py", line 91, in runcode
    exec(code, self.locals)
  File "<input>", line 1, in <module>
SystemError: <class 'qgis._core.QgsPoint'> returned a result with an error set

`

**Comment by nyalldawson** [comment]
Duplicate of #32478

**Comment by nyalldawson** [comment]
Sorry, that should have been #32443

**Comment by matheusfillipe** [comment]
Thanks @nyalldawson!  Any idea when this will be correct?

**Comment by nyalldawson** [comment]
It's already fixed, and has been backported for 3.10.1
